test/unit/durableclient-spec.ts

```diff
@@ -0,0 +1,423 @@
+        nock.restore();
+    });
+
+    describe("initialization", () => {
+        it("deserializes the durable client JSON schema correctly", () => {
+            // cast the JSON into the typed object
+            const input = JSON.parse(durableClientBindingInputJson) as OrchestrationClientInputData;
+            expect(input.taskHubName).to.be.equal(testTaskHubName);
+            expect(input.rpcBaseUrl).to.be.equal(testRpcBaseUrl);
+            expect(input.managementUrls).to.be.an("object");
+            expect(input.managementUrls.id).to.be.equal("INSTANCEID");
+            expect(input.managementUrls.statusQueryGetUri).to.startsWith(externalBaseUrl);
+        });
+    });
+
+    describe("createCheckStatusResponse()", () => {
+        it("does NOT reference the RPC endpoint", async () => {
+            const input = JSON.parse(durableClientBindingInputJson) as OrchestrationClientInputData;
+            const client = new DurableOrchestrationClient(input);
+            const request: HttpRequest = {
+                method: "GET",
+                url: `${externalOrigin}/api/Foo`,
+                headers: { },
+                query: { },
+                params: { },
+            };
+
+            const instanceId = "abc123";
+            const response = client.createCheckStatusResponse(request, instanceId);
+            const payload = response.body as HttpManagementPayload;
+
+            expect(payload).is.an("object");
+            expect(payload.id).to.be.equal(instanceId);
+
+            // Verify that even when using local RPC, we still expose the external
+            // URLs in the createCheckStatusResponse API result.
+            expect(payload.purgeHistoryDeleteUri).to.startWith(externalBaseUrl);
+            expect(payload.rewindPostUri).to.startWith(externalBaseUrl);
+            expect(payload.sendEventPostUri).to.startWith(externalBaseUrl);
+            expect(payload.statusQueryGetUri).to.startWith(externalBaseUrl);
+            expect(payload.terminatePostUri).to.startWith(externalBaseUrl);
+        });
+    });
+
+    describe("startNew()", () => {
+        it("uses the RPC endpoint", async () => {
+            const input = JSON.parse(durableClientBindingInputJson) as OrchestrationClientInputData;
+            const client = new DurableOrchestrationClient(input);
+
+            // The startNew() method should do a POST to http://127.0.0.1:17071/durabletask/orchestrators/MyFunction
+            const functionName = "MyFunction";
+            const expectedUrl = new URL(`${testRpcOrigin}/durabletask/orchestrators/${functionName}`);
+            const expectedHeaders = { reqheaders: { "Content-Type": "application/json" } };
+
+            const scope = nock(expectedUrl.origin, expectedHeaders)
+                .post(expectedUrl.pathname)
+                .reply(202, { id: "abc123" });
+
+            const result = await client.startNew(functionName);
+            expect(scope.isDone()).to.be.equal(true);
```

Comment:
Is this similar to verify() in C#?

Reply:
Not sure. I just copied this from other existing nock test code.

Reply:
It [looks like it is.](https://www.npmjs.com/package/nock#isdone). If it didn't, I would have recommended we did something to ensure the mock method was actually called, but it does!